torch/_inductor/runtime/runtime_utils.py

```diff
@@ -81,35 +80,74 @@ def do_bench(fn, fn_args, fn_kwargs, **kwargs):
         return do_bench_gpu(lambda: fn(*fn_args, **fn_kwargs), **kwargs)
 
 
-def do_bench_gpu(*args, **kwargs):
+def do_bench_gpu(
+    fn, warmup=25, rep=100, fast_flush=True, quantiles=(0.5,), return_mode="mean"
+):
     @functools.lru_cache(None)
-    def load_triton():
-        try:
-            # NB: Lazily load triton, as importing triton is slow
-            # see https://github.com/openai/triton/issues/1599
-            from triton.testing import do_bench as triton_do_bench
-        except ImportError as exc:
-            raise NotImplementedError("requires Triton") from exc
-
-        # triton PR https://github.com/openai/triton/pull/1513 change the
-        # quantile fields name from 'percentiles' to 'quantiles'
-        # and change the default value from (0.5, 0.2, 0.8) to None.
-        # This may break inductor since a caller expects a tuple may get a item.
-        #
-        # Add a wrapper to maintain the same behavior for inductor.
-        # Maybe we should have own implementation of this function?
-        return triton_do_bench, (
-            "quantiles"
-            if inspect.signature(triton_do_bench).parameters.get("quantiles")
-            is not None
-            else "percentiles"
+    def get_cache_size():
+        device = torch.cuda.current_device()
+        properties = torch.cuda.get_device_properties(device)
+        return properties.l2CacheSize
+
+    if fast_flush:
+        cache = torch.empty(int(256e6 // 4), dtype=torch.int, device="cuda")
+    else:
+        cache = torch.empty(int(256e6), dtype=torch.int8, device="cuda")
+
+    estimation_iters = 5
+    event_pairs = [
+        (
+            torch.cuda.Event(enable_timing=True),
+            torch.cuda.Event(enable_timing=True),
         )
+        for _ in range(estimation_iters)
+    ]
+    for start_event, end_event in event_pairs:
+        start_event.record()
+        cache.zero_()
+        fn()
+        end_event.record()
+    torch.cuda.synchronize()
+    estimate_ms = min(
+        [event_pair[0].elapsed_time(event_pair[1]) for event_pair in event_pairs]
+    )
+
+    if fast_flush:
+        cache = torch.empty(int(get_cache_size() // 4), dtype=torch.int, device="cuda")
```

Comment:
nit: could we `del cache` above before reassigning ? it doesn't practically matter, but the cache size is large enough so that it interferes with our benchmarking memory compression ratios for torchbench models. 

double checked myself that they would both be live at the same time:
```
class Foo():
    def __init__(self):
        print("Init")

    def __del__(self):
        print("Del")

a = Foo()
a = Foo()
```

```
Init
Init
Del
Del
```

Reply:
Sure; it would probably also help to manually delete the new cache at the end of benchmarking as well. I wonder if there is anything else like this messing with the memory compression ratios, might be worth looking at.